### PR TITLE
Added swagger support

### LINKED ISSUE #8: Create swagger schemas and document some endpoints

### PR DESCRIPTION
**Pull request by Astatium5**
TODO: 
Add schemas.yml file for referencing schemas in documentation

Closes #8

### CODE REVIEW

File: src/routes/group.js [hunk 1]
```diff
@@ -1,11 +1,45 @@
...
+ *
+ *      requestBody:
+ *        description:
+ *        required: true
+ *
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
```
Comment (Armintron) [inline src/routes/group.js:31]: Properties are missing; group creation requires a `creator` field, which represents the ObjectId of the user that owns the group.

File: src/routes/group.js [hunk 2]
```diff
@@ -1,11 +1,45 @@
...
+ *
+ *      produces:
+ *        - application/json
+ *      responses:
+ *        200:
+ *          description: OK
+ *        404:
+ *          description: Username/password combination is incorrect
```
Comment (Armintron) [inline src/routes/group.js:39]: There is no Username/Password combination for group registration.

File: src/routes/user.js [hunk 3]
```diff
@@ -1,19 +1,97 @@
...
+ *                  type: string
+ *                  example: Brown
+ *
+ *      produces:
+ *        - application/json
+ *      responses:
+ *        200:
+ *          description: OK
+ *        404:
+ *          description: Username/password combination is incorrect
```
Comment (Armintron) [inline src/routes/user.js:51]: You can't have an incorrect combination on user creation. An error would occur if the username is already taken as they are unique.

File: src/services/swagger.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        license: {
+          name: 'MIT',
+          url: 'https://spdx.org/licenses/MIT.html',
+        },
+      },
+      servers: [
+        {
+          url: 'http://localhost:5000/',
+          description: 'local server',
+        },
```
Comment (Armintron) [inline src/services/swagger.js:20]: The production server should also be added.

**Review by Armintron — CHANGES_REQUESTED** [review]
Overall looks good, just some simple changes and fixes from copy-pasting. Also, it would be nice if documentation could also be added for `groups/join/:inviteCode`, but we can always get to it later.

File: src/routes/group.js (hunk 1 above)
Comment (Astatium5) [reply src/routes/group.js:31]: Yes, that's in TODO. To references user, we need to create a user schema.

File: src/services/swagger.js (hunk 4 above)
Comment (Astatium5) [reply src/services/swagger.js:20]: I'll add it soon.

File: src/routes/user.js (hunk 3 above)
Comment (Astatium5) [reply src/routes/user.js:51]: Good catch. I'll fix it as well.

File: src/routes/group.js (hunk 2 above)
Comment (Astatium5) [reply src/routes/group.js:39]: It was just copied from user doc as a reference, because groups documentation requires some more work.

**Review by suneettipirneni — CHANGES_REQUESTED** [review]
I think I would want the landing page ('/') for the API to go right to swaggerhub since it's not used as an endpoint and makes it easier to access.

File: src/routes/group.js (hunk 1 above)
Comment (Astatium5) [reply src/routes/group.js:31]: I'll resolve the conversation, since it's in TODO

**Review by suneettipirneni — CHANGES_REQUESTED** [review]
Your responses don't give the JSON body that these requests give

**Review by suneettipirneni — APPROVED** [review]

**Pull request merged**